### PR TITLE
Update internal interface usage for FileExplorer integration

### PR DESCRIPTION
**Pull request by DefaultRyan**
## Summary of the pull request
This updates the Microsoft.Internal.Windows.DevHome.Helpers package version to the latest with the new interfaces for File Explorer integration. The associated interface implementation/wrappers are also updated here.

## References and relevant issues
As the OS changes with the new interfaces propagate and we near public preview, we should be able to deprecate and remove the old testing/prototyping interface.

## Validation steps performed
Built and installed this version of DevHome on Windows VMs:
- Only knew about the old testing interfaces.
- Updated builds that check for the new official interfaces.

### CODE REVIEW

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs
Comment (ssparach) [inline tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs:81]: Dev Home will not be the receiver of this exception but does it need to surface any error to the user? Thanks!

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/Program.cs
Comment (ssparach) [inline tools/Customization/DevHome.FileExplorerSourceControlIntegration/Program.cs:60]: Quick Question: Why is the same object instance sent twice as function argument? Is this related to the update in interface where one interface provides property information on folder and the other provides property information on files stored under the folder? Thanks!

File: extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj
Comment (dhoehna) [inline extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj:20]: Why is this version changing?

Comment (dhoehna) [inline extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj:20]: Does the GitHubextension and Azure Extension have PR's to upgrade to the same version?

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs
Comment (dhoehna) [inline tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs:81]: Why ArgumentOutOfRange?  Isn't rootFolderPath a string?

File: tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.ElevatedServer/packages.config
Comment (dhoehna) [inline tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.ElevatedServer/packages.config:3]: Consider moving the package reference to a common project.

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs
Comment (DefaultRyan) [reply tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs:81]: rootFolderPath is, indeed a string. I may have overthought this exception type. `ArgumentException` seems more appropriate.

File: extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj
Comment (DefaultRyan) [reply extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj:20]: It would be good for them to update when appropriate for them to do so, but it shouldn't be a requirement, since they're separate packages/extensions, yes?

Comment (DefaultRyan) [reply extensions/GitExtension/FileExplorerGitIntegration/FileExplorerGitIntegration.csproj:20]: There are some improvements in this latest version, but probably the most important one is that it updates LibGit2 to 1.7.2, which includes a couple security fixes and is recommended by the LibGit2 maintainers. https://github.com/libgit2/libgit2/releases/tag/v1.7.2

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/Program.cs
Comment (DefaultRyan) [reply tools/Customization/DevHome.FileExplorerSourceControlIntegration/Program.cs:60]: Yes, this is related to the interface update. One is for the legacy testing interface, and the other is for the new interface we'll use going forward. Rather than query at runtime for the existence of one of the interfaces (which could fail), I simply decided to provide two params to force the compiler to check for us. Once the helpers package can wean itself off the legacy testing interface, I'll delete the second param from that constructor and make the corresponding update here.

File: tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs
Comment (DefaultRyan) [reply tools/Customization/DevHome.FileExplorerSourceControlIntegration/SourceControlProvider.cs:81]: This will get called multiple times per file and would result in a bit of spamminess. I'm assuming that if we are failing to open the repository here, we'd fail to open the repository in the management or settings screen.

When we add telemetry, I'd like to add some to this case so we can track if this is something that users' machines hit. If so, we could figure out how and when to surface it vs log something for later. Might be something good to think about as we evolve the repo management user experience flows.

File: tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.ElevatedServer/packages.config
Comment (DefaultRyan) [reply tools/QuietBackgroundProcesses/DevHome.QuietBackgroundProcesses.ElevatedServer/packages.config:3]: That's a good tip for the feature branch to do before merging into main.

**Review by ssparach — APPROVED** [review]

**Pull request merged**
